crates/protect-ffi/src/encrypt_config.rs

```diff
@@ -0,0 +1,490 @@
+use super::Error;
```

Comment:
Copying this over from Proxy (until we extract the logic to somewhere shared). Main differences are that I've updated this to use protect-ffi's error type, fixed up a serialization bug (where `Identifier`s didn't use the `t` and `c` shorthands for `table` and `column`, and removed some functions that protect-ffi doesn't use.